pkg/utils/hex/hex.go

```diff
@@ -0,0 +1,55 @@
+package hex
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+)
+
+// EnsurePrefix adds the prefix (0x) to a given hex string.
+func EnsurePrefix(str string) string {
+	if !strings.HasPrefix(str, "0x") {
+		str = "0x" + str
+	}
+	return str
+}
+
+// ToBig parses the given hex string or panics if it is invalid.
+func ToBig(s string) *big.Int {
+	n, ok := new(big.Int).SetString(s, 16)
+	if !ok {
+		panic(fmt.Errorf(`failed to convert "%s" as hex to big.Int`, s))
```

Comment:
Where is this func used? Ideally we shouldn't be panicing in production code (especially a library) so I would hope that this can move to a test package, or be modified to return the error instead.

Reply:
Two places outside of tests:
- https://github.com/smartcontractkit/chainlink/blob/develop/core/services/keystore/keys/vrfkey/crypto.go#L22
- https://github.com/smartcontractkit/chainlink/blob/develop/core/services/vrf/v2/listener_v2_log_processor.go#L1185

Reply:
I'll update the function and fix the second case to check for an error. The first one will be modified to:

`FieldSize, _ := new(big.Int).SetString("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFC2F", 16)`

Reply:
Oh if it is in init() code, then panic is OK. We can wrap it with a generic `func must[T any](v T, err error) T` or something :shrug: